### ISSUE TITLE
Fix descriptions for Visit <page> links

### ISSUE THREAD

**Issue by atsansone** [flutter/website#10533]
From https://buganizer.corp.google.com/issues/337104253

### Observed Results

Navigating through the page specifically on Visit links, listen that all of them are announced the same description.

* [Screencast VoiceOver](http://screencast/cast/NTMyNDAxNzc5MDYxNTU1MnxlNjM4MTZjZi1lNQ)
* [Screencast ChromeVox](http://screencast/cast/NjMyOTQwMzYzMzA0MTQwOHwyMTY3ZDM4OS1jYQ)

### Expected Results

Ensure that Visit links should have a descriptive name to users example

- Visit accessibility site link.
- Visit Animation and Motion link...

### User Impact

Screen reader and Cognitive users are affected because all "Visit" links have the same description and are confused to them.

### Steps to Reproduce

1. Enabled VoiceOver using <kbd>Command</kbd> + <kbd>F5</kbd> also ChromeVox using <kbd>Ctrl</kbd> + <kbd>Alt</kbd> + <kbd>Z</kbd>
1. Open the following page: https://docs.flutter.dev/ui/widgets
1. Continue the linear navigation using the <kbd>Tab</kbd> key and reach the cards components.
1. Navigate through the links specifically on Visit links and listen to the results.

### GAR Assessment

[Flutter Docs](https://a11ytracker.corp.google.com/products/c2fdcf18-53d2-475d-b8b0-e5dedf8cb900?e=ProductLabels)

### GAR Details

[1.1. Do all non-text content and user interface components have meaningful accessible names and alternative text?](go/GAR-web#accessible-names)

### Environment Details

Machine: MAC/ChromeBook
Browser: Chrome Version 120.0.6099.272 (Official Build) (64-bit)
Screen reader: VoiceOver / ChromeVox.
Bug filling time: 15 mins.
